src/ui/projectOutline/projectOutline.ts

```diff
@@ -35,6 +35,56 @@ abstract class BaseNode {
     abstract getOrderTuple(): string[];
 }
 
+/**
+ * Filter node that appears at the top of the tree to show and edit the current filter
+ */
+export class FilterNode extends BaseNode {
+    constructor(private searchTerm: string) {
+        super('cmake.outline.filter');
+    }
+
+    getChildren(): BaseNode[] {
+        return [];
+    }
+
+    getOrderTuple(): string[] {
+        return ['0-filter']; // Ensures it appears first
+    }
+
+    updateSearchTerm(term: string) {
+        this.searchTerm = term;
+    }
+
+    getTreeItem(): vscode.TreeItem {
+        const item = new vscode.TreeItem(
+            this.searchTerm ? `Filter: "${this.searchTerm}"` : 'Filter: (none)',
+            vscode.TreeItemCollapsibleState.None
+        );
+        item.id = this.id;
+        item.tooltip = this.searchTerm ?
+            'Click to edit filter or clear with X button' :
+            'Click to set a filter for the project outline';
+        item.contextValue = 'filter';
+
+        // Add appropriate icon and command
+        if (this.searchTerm) {
+            item.iconPath = new vscode.ThemeIcon('filter-filled');
+            item.description = `(${this.searchTerm.length} chars)`;
+        } else {
+            item.iconPath = new vscode.ThemeIcon('filter');
+            item.description = 'Click to filter';
```

Comment:
We need to localize this, so it'll look something like this `localize('click.to.filter', "Click to filter")`